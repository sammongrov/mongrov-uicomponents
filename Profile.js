import React, { Component } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { styles } from 'react-native-theme';
import moment from 'moment';
import { Text, Avatar, Icon } from '@ui/components';
import PropTypes from 'prop-types';
import { iOSColors } from 'react-native-typography';

export default class Profile extends Component {
  state = {};

  renderIcon() {
    const { onPress, showIcon, iconName, iconSize, iconColor } = this.props;
    if (showIcon) {
      return (
        <TouchableOpacity style={styles.profileIconContainer} onPress={onPress}>
          <Icon
            size={iconSize}
            reverse
            name={iconName}
            color={iconColor}
            style={styles.profileIconSpace}
          />
        </TouchableOpacity>
      );
    }
    return null;
  }

  renderMembersData() {
    const { totalMembers, onlineMembers } = this.props;
    if (totalMembers && onlineMembers) {
      return (
        <Text style={[styles.subtextBlack1, styles.textContainer]}>
          {`${totalMembers} members (${onlineMembers}) online`}
        </Text>
      );
    }
    return null;
  }

  renderProfileUserName() {
    const { profileUname } = this.props;
    if (profileUname) {
      return <Text style={[styles.subtextBlackbody, styles.textContainer]}>{profileUname}</Text>;
    }
    return null;
  }

  renderProfileAnnouncement() {
    const { profileAnnouncement } = this.props;
    if (profileAnnouncement) {
      return (
        <Text style={[styles.subtextBlackbody, styles.textContainer]}>{profileAnnouncement}</Text>
      );
    }
    return null;
  }

  renderStatusDetail() {
    const { showStatusText, statusColor } = this.props;
    if (showStatusText) {
      return (
        <View style={[styles.memberSubView]}>
          <View style={[styles.statusCircle, { backgroundColor: statusColor }]} />
          <Text style={[styles.subtextBlackbody]}>{showStatusText}</Text>
        </View>
      );
    }
    return null;
  }

  renderEmailDetail() {
    const { profileMailId } = this.props;
    if (profileMailId) {
      return (
        <View style={[styles.memberSubView]}>
          <Icon
            name="email-outline"
            type="material-community"
            style={styles.profileSubIconStyle}
            size={17}
            color={iOSColors.gray}
          />
          <Text style={[styles.subtextBlackbody]}>{profileMailId}</Text>
        </View>
      );
    }
    return null;
  }

  renderLoginTime() {
    const { lastLoginTime } = this.props;
    if (lastLoginTime) {
      return (
        <View style={[styles.memberSubView]}>
          <Icon
            name="clock"
            type="material-community"
            style={styles.profileSubIconStyle}
            size={17}
            color={iOSColors.gray}
          />
          <Text style={[styles.subtextBlackbody]}>
            {`Last Login at ${moment(new Date(lastLoginTime)).format('hh:mm A')}`}
          </Text>
        </View>
      );
    }
    return null;
  }

  render() {
    const { avatarName, avatarUrl, avatarSize, profileTitle, profileStyle } = this.props;
    return (
      <View style={[styles.paddingVertical30, styles.alignJustifyCenter, profileStyle]}>
        <View>
          <Avatar avatarUrl={avatarUrl} avatarName={avatarName} avatarSize={avatarSize} />
          {this.renderIcon()}
        </View>
        <Text style={[styles.groupTitleText, styles.textContainer]}>{profileTitle}</Text>
        {this.renderProfileUserName()}
        {this.renderProfileAnnouncement()}
        {this.renderMembersData()}
        {this.renderStatusDetail()}
        {this.renderEmailDetail()}
        {this.renderLoginTime()}
      </View>
    );
  }
}

Profile.propTypes = {
  profileStyle: PropTypes.shape(),
  iconColor: PropTypes.string,
  avatarUrl: PropTypes.string,
  avatarName: PropTypes.string,
  iconName: PropTypes.string,
  avatarSize: PropTypes.number,
  iconSize: PropTypes.number,
  showIcon: PropTypes.bool,
  profileTitle: PropTypes.string,
  onPress: PropTypes.func,
  profileUname: PropTypes.string,
  showStatusText: PropTypes.string,
  profileMailId: PropTypes.string,
  lastLoginTime: PropTypes.string,
  statusColor: PropTypes.string,
  totalMembers: PropTypes.string,
  onlineMembers: PropTypes.string,
  profileAnnouncement: PropTypes.string,
};

Profile.defaultProps = {
  profileStyle: null,
  iconColor: null,
  iconName: 'camera-iris',
  avatarUrl: null,
  avatarName: '',
  avatarSize: 60,
  iconSize: 30,
  showIcon: false,
  profileTitle: '',
  onPress: () => {},
  profileUname: '',
  showStatusText: '',
  profileMailId: '',
  lastLoginTime: '',
  statusColor: null,
  totalMembers: '',
  onlineMembers: '',
  profileAnnouncement: '',
};
