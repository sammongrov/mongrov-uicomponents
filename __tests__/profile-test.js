import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Alert } from 'react-native';
import Profile from '../Profile';

configure({ adapter: new Adapter() });

const onPress = () => Alert.alert('message');
const title = { showIcon: true };

const mockMath = global.Math;
mockMath.random = () => 0.5;
global.Math = mockMath;

it('Profile renders correctly', () => {
  const tree = renderer
    .create(
      <Profile
        title={title}
        avatarName="Avatar"
        iconName="camera-iris"
        avatarSize={60}
        iconSize={30}
        onPress={onPress}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('show props test', () => {
  const wrapper = shallow(<Profile showIcon={true} />);
  expect(wrapper.instance().props.showIcon).not.toBeNull();
});

it('member data renders correctly', () => {
  const tree = renderer
    .create(<Profile totalMembers="7" onlineMembers="5" profileUname="john" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('username renders correctly', () => {
  const tree = renderer.create(<Profile profileUname="john" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('announcement renders correctly', () => {
  const tree = renderer.create(<Profile profileAnnouncement="hello" />).toJSON();
  expect(tree).toMatchSnapshot();
});
it('status text renders correctly', () => {
  const tree = renderer.create(<Profile showStatusText="online" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('email information renders correctly', () => {
  const tree = renderer.create(<Profile profileMailId="john@gmail.com" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('login time renders correctly', () => {
  const tree = renderer.create(<Profile lastLoginTime="5.30" />).toJSON();
  expect(tree).toMatchSnapshot();
});
