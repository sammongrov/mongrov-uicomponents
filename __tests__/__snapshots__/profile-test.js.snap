// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#7597ad",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          A
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`announcement renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    hello
  </Text>
</View>
`;

exports[`email information renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#8E8E93",
            "fontSize": 17,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="material-community"
    >
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Regular",
          },
          Array [
            undefined,
          ],
        ]
      }
    >
      john@gmail.com
    </Text>
  </View>
</View>
`;

exports[`login time renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#8E8E93",
            "fontSize": 17,
          },
          undefined,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      type="material-community"
    >
      肋
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Regular",
          },
          Array [
            undefined,
          ],
        ]
      }
    >
      Last Login at 12:00 AM
    </Text>
  </View>
</View>
`;

exports[`member data renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    john
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    7 members (5) online
  </Text>
</View>
`;

exports[`status text renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
  <View
    style={
      Array [
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "backgroundColor": null,
          },
        ]
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "OpenSans-Regular",
          },
          Array [
            undefined,
          ],
        ]
      }
    >
      online
    </Text>
  </View>
</View>
`;

exports[`username renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
      null,
    ]
  }
>
  <View>
    <View
      style={
        Array [
          undefined,
          Object {
            "position": "relative",
          },
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#2E88FF",
            "borderRadius": 30,
            "height": 60,
            "opacity": 1,
            "width": 60,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "OpenSans-Regular",
        },
        Array [
          undefined,
          undefined,
        ],
      ]
    }
  >
    john
  </Text>
</View>
`;
